File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/flutterobject/ArcgisNativeObjectFactory.java
package com.valentingrigorean.arcgis_maps_flutter.flutterobject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface ArcgisNativeObjectFactory {
    @NonNull
    NativeObject createNativeObject(@NonNull String objectId, @NonNull String type, @Nullable Object arguments, @NonNull NativeMessageSink messageSink);
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/flutterobject/ArcgisNativeObjectFactory.kt
package com.valentingrigorean.arcgis_maps_flutter.flutterobject

interface ArcgisNativeObjectFactory {
    fun createNativeObject(
        objectId: String,
        type: String,
        arguments: Any?,
        messageSink: NativeMessageSink
    ): NativeObject
}